packages/formik/src/Formik.tsx

```diff
@@ -364,6 +364,16 @@ export function useFormik<Values extends FormikValues = FormikValues>({
     }
   );
 
+  React.useEffect(() => {
+    if (
+      validateOnMount &&
+      isMounted.current === true &&
+      isEqual(initialValues.current, props.initialValues)
```

Comment:
Why we need perform validation only if form values equal initial values? What about the case when the values on form are empty?

Reply:
I don't know what you mean about empty. The bug this fixes is that the form wouldn't revalidate if initialValues prop stays the same. There is already validation happening in the other useEffect I've modified (it doesn't display in the diff) that handles the case where it does change. Why can't it be a single useEffect performing validation? Two reasons: the first was the need for #2507 (revalidating when enableReinitialize is true). The second I've covered in my comment above.